### PR TITLE
[FLINK-36075] Add pagination partitioning strategy

### PR DESCRIPTION
**Pull request by yuxiqian**
This closes FLINK-36075 by:

* Adding new enum value `pagination` for `scan.partition.strategy` option
* Adding `scan.partition.record-size` option to specify the pagination size for each split

Marked it as a draft since more discussion is needed over the API design.

### CODE REVIEW

**Comment by yuxiqian** [comment]
Seems CI is failing due to connectivity issues with Nexus repository.

File: flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/enumerator/splitter/MongoPaginationSplitter.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    private MongoPaginationSplitter() {}
+
+    public static Collection<MongoScanSourceSplit> split(MongoSplitContext splitContext) {
+        int partitionRecordSize = splitContext.getReadOptions().getPartitionRecordSize();
+        Preconditions.checkArgument(
+                partitionRecordSize > 0,
+                "The partition record size must be set to a positive integer.");
+
+        int numberOfPartitions =
+                (int) (Math.ceil(splitContext.getCount() / (double) partitionRecordSize));
```
Comment (Jiabao-Sun) [inline flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/enumerator/splitter/MongoPaginationSplitter.java:53]: Thanks @yuxiqian for the great work. Looks good overall.

A suggestion is that we might not need to set the default value of `scan.partition.record-size` to 0, but rather leave it without a default value. This is because having 0 as the default value can be somewhat misleading. I prefer that when the user does not explicitly set the `scan.partition.record-size` config option, the number of documents per partition should be calculated based on `scan.partition.size` and `avgObjSize`, similar to the `MongoSampleSplitter`. When the user explicitly sets the `scan.partition.record-size` config option, the number of documents per partition should be determined by this parameter.

https://github.com/apache/flink-connector-mongodb/blob/e0a04d0e64e7e8cfa073ab92abbc2c6c0ae20830/flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/enumerator/splitter/MongoSampleSplitter.java#L83-L97

Additionally, it would be best for `MongoPaginationSplitter` to provide unit tests similar to `MongoPaginationSplitterTest` to verify edge cases.

**Review by Jiabao-Sun — APPROVED** [review]
Thanks @yuxiqian for this great work and @GOODBOY008 for the review.
LGTM
